src/vs/workbench/common/editor/editorGroupModel.ts

```diff
@@ -739,14 +742,14 @@ export class EditorGroupModel extends Disposable implements IEditorGroupModel {
 	private doSetSelection(newActiveEditor: EditorInput, activeEditorIndex: number, inactiveSelectedEditors: EditorInput[]): void {
 		this.doSetActive(newActiveEditor, activeEditorIndex);
 
-		for (const candidate of inactiveSelectedEditors) {
+		for (const candidate of inactiveSelectedEditors) { // TODO should this move up to the public API? the idea is that we do not trust editor inputs from public API but internal is fine
 			const editor = this.findEditor(candidate)?.[0];
 			if (editor && !this.isSelected(editor)) {
 				this.selection.push(editor);
 			}
 		}
 
-		// Event
+		// Event TODO what if selection did not change?
```

Comment:
@benibenj this seems to always emit a selection change event even if nothing changed?